File: golabs/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/java/Ch02-Schemas-and-Indexes/src/SipOfRiak/OrderSummaryItem.java
package SipOfRiak;

public class OrderSummaryItem
{
    public OrderSummaryItem() { }
    public OrderSummaryItem(Order parentOrder) {
        OrderId = parentOrder.OrderId;
        Total = parentOrder.Total;
        OrderDate = parentOrder.OrderDate;
    }

    public long OrderId;
    public Double Total;
    public String OrderDate;
}

File: golabs/godata/go-gumball-riak/gumball-riak/src/github.com/basho/taste-of-riak/csharp/Ch02-Schemas-and-Indexes/OrderSummaryItem.cs
namespace Ch02_Schemas_and_Indexes
{
    using System;

    public class OrderSummaryItem
    {
        private readonly uint orderId;
        private readonly decimal total;
        private readonly DateTime orderDate;

        public OrderSummaryItem(Order parentOrder)
        {
            if (parentOrder == null)
            {
                throw new ArgumentNullException("parentOrder");
            }

            this.orderId = parentOrder.Id;
            this.total = parentOrder.Total;
            this.orderDate = parentOrder.OrderDate;
        }

        public uint OrderId
        {
            get { return orderId; }
        }

        public decimal Total
        {
            get { return total; }
        }

        public DateTime OrderDate
        {
            get { return orderDate; }
        }
    }
}
